integrate/simpsons.go

```diff
@@ -0,0 +1,64 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package integrate
+
+import "sort"
+
+// Simpsons estimates the integral of a function f
+//  \int_a^b f(x) dx
+// from a set of evaluations of the function using the extended Simpson's rule.
+// The extended Simpson's rule
+// fits a polynomial approximation to the function given
+// two or more adjacent Simpson's rule calculations with shared endpoints,
+// by taking the mean of composite Simpson's 1/8 rule
+// and composite Simpson's 3/8 rule applied to overlapping segments
+// to estimate
+//  \int_x[1]^x[n] f(x) dx
+// where f[i] is the value of the function at x[i].
+// The form and mathematical formulae of extended Simpson's rule can be found at:
+// https://en.wikipedia.org/wiki/Simpson's_rule
+//
+// The (x,f) input data points must be sorted along x.
+// One can use github.com/gonum/stat.SortWeighted to do that.
+// The x and f slices must be of equal length and have length > 1.
+func Simpsons(x, f []float64) float64 {
```

Comment:
The signature makes it appear as if the interior elements of `x` matter but they do not. I think the function as it is, if we decide to accept this, should take the endpoints of the integration interval, the number of subintervals and the function.

Reply:
f should be ordered. Ordering is determined by the ordering of values in x. f is not a function. The subintervals are each element of f.

Reply:
Regarding `f`, you're right, I was stuck in thinking in terms of `integrate.Fixed` API. Regarding `x`, I'm not convinced, it's too easy to make a mistake. Either the algorithm should use the elements of `x` or the signature should be different.

Reply:
I've made updates to make use of elements of x